apps/settings/lib/SetupChecks/CheckServerResponseTrait.php

```diff
@@ -93,6 +93,10 @@ protected function getTestUrls(string $url, bool $isRootRequest = false): array
 	protected function normalizeUrl(string $url, bool $removeWebroot): string {
 		if ($removeWebroot) {
 			$segments = parse_url($url);
+			if (!isset($segments['scheme']) || !isset($segments['host'])) {
+				throw new \InvalidArgumentException('URL is missing scheme or host');
+			}
+
```

Comment:
```suggestion
                if (!filter_var($url, FILTER_VALIDATE_URL)) {
                        throw new \InvalidArgumentException('URL ($url) is invalid - Please verify syntax of all URLs / domains / IP addresses in your config');
                }
		if ($removeWebroot) {
			$segments = parse_url($url);
```

Probably fine either way, but I think using `filter_var` is safer / more likely to catch edge cases.

Also adding the url in the error will help the operator track down the culprit faster.

Reply:
Fine by me, but in that case, this should be into `master` too/first.

Reply:
I guess if we change the approach here, we'll need to do the same in `master`: 

https://github.com/nextcloud/server/blob/935f0d27acfdbb60b11924688aeaaeced62a1e6e/lib/public/SetupCheck/CheckServerResponseTrait.php#L150-L161

Reply:
On it... (in `master`)

Reply:
#49379